### PR TITLE
Ports and Adapts TGUI strip panel from TG

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Ports https://github.com/tgstation/tgstation/pull/57889.

- [x] fix no custom PDA background sprite
- [x] possibly redo ALL the background sprites
- [x] remove all the Topic() calls for inventory
- [x] fix pickpocket gloves
- [x] make so you can actually apply a collar to any simple animal again? (i forget what the restrictions were)
- [x] restrict ui_state/ui_status on the strip menu
- [x] Fix PDA showing up as non-obscured
- [x] Test stuff like muzzles that need IDs to take off
- [x] Make headpockets properly hidden when your mask is hidden (like from a space helmet)
- [x] Check if the strip menu is open before opening a new one
- [x] strip_accessory
- [x] make ghosts not able to start stripping people
- [ ] Make the "R" and "L" not disappear from hands when the person is holding something

Fixes #24618

## Why It's Good For The Game

Our current strip menu feels very outdated to the rest of the game. However, we HAVE THE TECHNOLOGY. Also removes a fuck-ton of shitcode in exchange for new shitcode. Also, it looks a lot better, and is more consistent with how the rest of the HUD is laid out on the screen.

## Images of changes

image

Support for multiple alternate actions on an item
image

Support for skrell headpockets
image

Tooltips for empty slots, items, and alternate actions!
image

Accessible items, but that are hidden (like pockets or a PDA) are shown with a crossed-eye symbol.
image

Items that are completely obscured (like by wearing a space suit over them), are shown with the crossed-circle symbol.
image

Items that are NODROP, or cannot be stripped for another reason, have a transparent background.
image

Supports handcuffs (as shown in the first image) and legcuffs as well
image

## Testing

- [x] modsuit testing?
- [x] internals testing
- [x] test stripping aliens, corgis, and parrots (poly)
- [x] test handcuffs
- [x] test legcuffs
- [x] Abstract items are hidden

## Changelog
:cl:
add: Stripping menu is now tgui.
tweak: Adding/Removing items from the inventories of corgis and parrots are no longer instant.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+  const { act, data } = useBackend<StripMenuData>(context);
+
+  const gridSpots = new Map<GridSpotKey, string>();
+  for (const key of Object.keys(data.items)) {
+    gridSpots.set(SLOTS[key].gridSpot, key);
+  }
+
+  return (
+    <Window title={`Stripping ${data.name}`} width={360} height={390}>
+      <Window.Content>
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:257]:
```suggestion
      <Window.Content style={{'background-color': 'rgba(0, 0, 0, 0.5)'}}>
```

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+export const StripMenu = (props, context) => {
+  const { act, data } = useBackend<StripMenuData>(context);
+
+  const gridSpots = new Map<GridSpotKey, string>();
+  for (const key of Object.keys(data.items)) {
+    gridSpots.set(SLOTS[key].gridSpot, key);
+  }
+
+  return (
+    <Window title={`Stripping ${data.name}`} width={360} height={390}>
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:256]:
```suggestion
    <Window title={`Stripping ${data.name}`} width={360} height={390} theme="nologo">
```

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+                          style={{
+                            background: item?.interacting
+                              ? 'hsl(39, 73%, 30%)'
+                              : '#32475e',
+                            position: 'relative',
+                            width: '100%',
+                            height: '100%',
+                            padding: 0,
+                          }}
+                        >
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:361]:
```suggestion
                        <Button
                          onClick={() => {
                            act('use', {
                              key: keyAtSpot,
                            });
                          }}
                          fluid
                          color={item?.interacting ? 'average' : 'translucent'}
                          tooltip={tooltip}
                          style={{
                            position: 'relative',
                            width: '100%',
                            height: '100%',
                            padding: 0,
                          }}
                        >
```

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+                            tooltip={alternateAction.text}
+                            width="1.8em"
+                            style={{
+                              background: 'rgba(0, 0, 0, 0.6)',
+                              position: 'absolute',
+                              bottom: 0,
+                              right: 0,
+                              'z-index': 2,
+                            }}
+                          >
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:400]:
```suggestion
                          <Button
                            onClick={() => {
                              act('alt', {
                                key: keyAtSpot,
                              });
                            }}
                            color="blue"
                            tooltip={alternateAction.text}
                            width="1.8em"
                            style={{
                              position: 'absolute',
                              bottom: 0,
                              right: 0,
                              'z-index': 2,
                            }}
                          >
```

**Review by AyIong — COMMENTED** [review]
Just a little design suggestion :з

**Comment by dj-34** [comment]
Could we have a little tooltip window that shows object name on hover?

File: code/__DEFINES/dcs/signals.dm [hunk 5]
```diff
@@ -1010,3 +1010,6 @@
 
 /// from /obj/structure/cursed_slot_machine/determine_victor() when someone finally wins.
 #define COMSIG_GLOB_CURSED_SLOT_MACHINE_WON "cursed_slot_machine_won"
+
+///from of mob/MouseDrop(): (/atom/over, /mob/user)
+#define COMSIG_DO_MOB_STRIP "cursed_slot_machine_won"
```
Comment (dj-34) [inline code/__DEFINES/dcs/signals.dm:1015]: Is it supposed to be "cursed_slot_machine_won" for COMSIG_DO_MOB_STRIP?

File: code/__DEFINES/mob_defines.dm [hunk 6]
```diff
@@ -173,7 +173,7 @@
 ///The shock doesn't stun.
 #define SHOCK_NOSTUN 	(1<<3)
 
-#define POCKET_STRIP_DELAY			40	//time taken (in deciseconds) to search somebody's pockets
+#define POCKET_STRIP_DELAY			4 SECONDS	//time taken to search somebody's pockets
 
 #define DEFAULT_ITEM_STRIP_DELAY		40  //time taken (in deciseconds) to strip somebody
 #define DEFAULT_ITEM_PUTON_DELAY		20  //time taken (in deciseconsd) to reverse-strip somebody
```
Comment (dj-34) [inline code/__DEFINES/mob_defines.dm:179]: Probably could do the same for this
```suggestion
#define POCKET_STRIP_DELAY			4 SECONDS	//time taken to search somebody's pockets

#define DEFAULT_ITEM_STRIP_DELAY		4 SECONDS  //time taken to strip somebody
#define DEFAULT_ITEM_PUTON_DELAY		2 SECONDS  //time taken to reverse-strip somebody
```

File: code/__DEFINES/dcs/signals.dm (hunk 5 above)
Comment (Contrabang) [reply code/__DEFINES/dcs/signals.dm:1015]: uhhh, no I just forgot

**Comment by dj-34** [comment]
I would like to see items with NODROP have a dark red background rather than transparent (for ease of understanding), as Aylong mentioned in Discord.

**Comment by Contrabang** [comment]
Yeah, the annoying thing is that buttons don't really work very well like that, with custom colors. I'm honestly fine with it as now, and changes can always be made post-merge

**Review by warriorstar-orion — APPROVED** [review]
The DM looks fine, I gave the TGUI changes a once-over but you should probably get a second set of eyes on the TypeScript.

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+                    content = (
+                      <Icon
+                        name={
+                          item.obscured === ObscuringLevel.Completely
+                            ? 'ban'
+                            : 'eye-slash'
+                        }
+                        size={3}
+                        ml={0}
+                        mt={2.75}
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:347]:
```suggestion
                        mt={2.5}
```
Icon is slightly off-centre

**Comment by Scribble-Sheep** [comment]
Yoooo, no more staring at the strip screen and just getting lost in all the text! Nice!

File: tgui/packages/tgui/interfaces/StripMenu.tsx [hunk 8]
```diff
@@ -281,6 +282,13 @@ export const StripMenu = (props, context) => {
     return 'translucent';
   };
 
+  const disable_background_hover = (item) => {
+    if (item && item.cantstrip) {
+      return 'transparent';
```
Comment (AyIong) [inline tgui/packages/tgui/interfaces/StripMenu.tsx:287]: You can puy any color, but why you doesnt?

Comment (Contrabang) [reply tgui/packages/tgui/interfaces/StripMenu.tsx:287]: I prefer it with no color

**Review by S34NW — APPROVED** [review]

**Pull request merged**